### PR TITLE
Add YTDownloader plugin

### PR DESCRIPTION
**Pull request by DeepVoyager253**
I made a Youtube downloader plugin, that allows you to type in: -yt URL and will let you choose between downloading and mp3 or an mp4 of the video. if there are any issues that prevent my plugin from getting in, i would like to know how to correct them as i am dedicated to make this project work. Please keep in mind, that it is still in pretty early development.

i.e. I am pretty new to flow launcher plugin development, and if i have any problems with the workflow or plugin.json i would like you to tell me so i can fix it.

i.e.2 if you have any issues with running the plugin, please let me know how i can fix it because for me it works perfectly.

### CODE REVIEW

File: plugins.json [hunk 1]
```diff
@@ -1339,5 +1339,19 @@
...
+    },
+    {
+    "ID": "d58fbe34-bce7-11ed-afa1-0242ac120002",
+    "ActionKeyword": "-yt",
+    "Name": "Youtube Downloader",
+    "Description": "A youtube downloader",
+    "Author": "LordOfTheEel",
+    "Version": "1.0.0-alpha",
+    "Language": "python",
+    "Website": "n/a",
```
Comment (Garulf) [inline plugins.json:1351]:
```suggestion
    "Website": "https://github.com/DeepVoyager253/FlowLauncherYTDownload",
```

Perfectly acceptable to use the GitHub page here.

**Review by Garulf — CHANGES_REQUESTED** [review]

File: plugins.json [hunk 2]
```diff
@@ -1339,5 +1339,19 @@
...
+    "IcoPath": "cat.png",
+    "ExecuteFileName": "main.py",
+    "UrlSouceCode": "https://github.com/DeepVoyager253/FlowLauncherYTDownload"
```
Comment (Garulf) [inline plugins.json:1354]:
```suggestion
    "UrlSouceCode": "https://github.com/DeepVoyager253/FlowLauncherYTDownload",
```

Looks like a missing `,`

**Review by Garulf — CHANGES_REQUESTED** [review]

**Comment by DeepVoyager111** [comment]
After you've fixed it, can you re-check it?

File: plugins.json [hunk 3]
```diff
@@ -1339,5 +1339,19 @@
...
-]
\ No newline at end of file
+    },
+    {
+    "ID": "d58fbe34-bce7-11ed-afa1-0242ac120002",
+    "ActionKeyword": "-yt",
+    "Name": "Youtube Downloader",
+    "Description": "A youtube downloader",
+    "Author": "LordOfTheEel",
+    "Version": "1.0.0-alpha",
```
Comment (Garulf) [inline plugins.json:1349]:
```suggestion
    "Version": "1.0.0-beta",
```

File: plugins.json [hunk 4]
```diff
@@ -1339,5 +1339,19 @@
...
-]
\ No newline at end of file
+    },
+    {
+    "ID": "d58fbe34-bce7-11ed-afa1-0242ac120002",
+    "ActionKeyword": "-yt",
+    "Name": "Youtube Downloader",
+    "Description": "A youtube downloader",
+    "Author": "LordOfTheEel",
+    "Version": "1.0.0-beta",
```
Comment (Garulf) [inline plugins.json:1349]:
```suggestion
    "Version": "1.0.0-alpha",
```

File: plugins.json [hunk 5]
```diff
@@ -1339,5 +1339,19 @@
...
-]
\ No newline at end of file
+    },
+    {
+    "ID": "d58fbe34-bce7-11ed-afa1-0242ac120002",
+    "ActionKeyword": "-yt",
+    "Name": "Youtube Downloader",
+    "Description": "A youtube downloader",
+    "Author": "LordOfTheEel",
+    "Version": "v1.0.0-beta",
```
Comment (Garulf) [inline plugins.json:1349]:
```suggestion
    "Version": "vv1.0.0-beta",
```

File: plugins.json [hunk 6]
```diff
@@ -1339,5 +1339,19 @@
...
+    "Language": "python",
+    "Website": "https://github.com/DeepVoyager253/FlowLauncherYTDownload",
+    "IcoPath": "https://cdn.jsdelivr.net/gh/DeepVoyager253/FlowLauncherYTDownload@main/cat.png?raw=true",
+    "ExecuteFileName": "main.py",
+    "UrlSouceCode": "https://github.com/DeepVoyager253/FlowLauncherYTDownload",
+    "UrlDownload": "https://github.com/DeepVoyager253/FlowLauncherYTDownload/archive/refs/heads/main.zip"
```
Comment (Garulf) [inline plugins.json:1355]:
```suggestion
    "UrlDownload": "https://github.com/DeepVoyager253/FlowLauncherYTDownload/releases/download/v1.0.0-beta/Flow.Launcher.Plugin.YTDownloader.zip"
```

**Review by DeepVoyager253 — COMMENTED** [review]
t

**Review by DeepVoyager253 — COMMENTED** [review]
t

**Review by DeepVoyager253 — COMMENTED** [review]
t

**Review by DeepVoyager253 — COMMENTED** [review]
t

**Review by DeepVoyager253 — COMMENTED** [review]
t

**Review by DeepVoyager253 — COMMENTED** [review]
t

**Review by DeepVoyager253 — COMMENTED** [review]
v

**Review by DeepVoyager253 — COMMENTED** [review]
1

**Review by DeepVoyager253 — COMMENTED** [review]
111

**Review by DeepVoyager253 — COMMENTED** [review]
1

**Comment by Garulf** [comment]
After choosing the "Download your video in full quailty" option I get this error:

```
Please open new issue in: https://github.com/Flow-Launcher/Flow.Launcher/issues/new
1. upload log file: C:\Users\Garulf\AppData\Roaming\FlowLauncher\Logs\1.14.0\2023-03-09.txt
2. copy below exception message

Flow Launcher version: 1.14.0
OS Version: Microsoft Windows NT 10.0.19045.0
IntPtr Length: 8
x64: True

Python Path: C:\Users\Garulf\AppData\Local\Programs\Python\Python311\pythonw.exe
Node Path: 
Date: 03/09/2023 12:11:06
Exception:
System.IO.InvalidDataException: Traceback (most recent call last):
  File "C:\Users\Garulf\AppData\Roaming\FlowLauncher\Plugins\YTDownloader\main.py", line 84, in <module>
    YoutubeDownload()
  File "C:\Users\Garulf\AppData\Roaming\FlowLauncher\Plugins\YTDownloader\lib\flowlauncher\FlowLauncher.py", line 33, in __init__
    results = request_method(*request_parameters)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Garulf\AppData\Roaming\FlowLauncher\Plugins\YTDownloader\main.py", line 46, in download_audio
    yt_video = pytube.YouTube(url)
               ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Garulf\AppData\Roaming\FlowLauncher\Plugins\YTDownloader\lib\pytube\__main__.py", line 71, in __init__
    self.video_id = extract.video_id(url)
                    ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Garulf\AppData\Roaming\FlowLauncher\Plugins\YTDownloader\lib\pytube\extract.py", line 133, in video_id
    return regex_search(r"(?:v=|\/)([0-9A-Za-z_-]{11}).*", url, group=1)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Garulf\AppData\Roaming\FlowLauncher\Plugins\YTDownloader\lib\pytube\helpers.py", line 129, in regex_search
    raise RegexMatchError(caller="regex_search", pattern=pattern)
pytube.exceptions.RegexMatchError: regex_search: could not find match for (?:v=|\/)([0-9A-Za-z_-]{11}).*

   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 300
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 300
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.<>c__DisplayClass19_0.<<ParseResults>b__0>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 139
--- End of stack trace from previous location ---
   at Flow.Launcher.ViewModel.MainViewModel.OpenResultAsync(String index) in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 277
   at CommunityToolkit.Mvvm.Input.AsyncRelayCommand.AwaitAndThrowIfFailed(Task executionTask)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   ```

File: plugins.json [hunk 7]
```diff
@@ -1339,5 +1339,19 @@
...
+    "ID": "d58fbe34-bce7-11ed-afa1-0242ac120002",
+    "ActionKeyword": "-yt",
+    "Name": "Youtube Downloader",
+    "Description": "A youtube downloader",
+    "Author": "LordOfTheEel",
+    "Version": "v1.0.0-beta",
+    "Language": "python",
+    "Website": "https://github.com/DeepVoyager253/FlowLauncherYTDownload",
+    "IcoPath": "https://cdn.jsdelivr.net/gh/DeepVoyager253/FlowLauncherYTDownload@main/cat.png?raw=true",
+    "ExecuteFileName": "main.py",
```
Comment (Garulf) [inline plugins.json:1353]:
```suggestion
```

File: plugins.json [hunk 8]
```diff
@@ -1339,5 +1339,19 @@
         "IcoPath": "https://cdn.jsdelivr.net/gh/kenty02/Flow.Launcher.Plugin.JetBrainsIDEProjects/Flow.Launcher.Plugin.JetBrainsIDEProjects/icon.png",
         "DateAdded": "2023-02-19T21:47:51Z",
         "LatestReleaseDate": "2023-03-01T00:58:12Z"
-    }
-]
\ No newline at end of file
+    },
+    {
+    "ID": "d58fbe34-bce7-11ed-afa1-0242ac120002",
+    "ActionKeyword": "-yt",
```
Comment (Garulf) [inline plugins.json:1345]:
```suggestion
```

**Review by Garulf — APPROVED** [review]

**Pull request merged**